mongomock/filtering.py

```diff
@@ -61,6 +57,10 @@ def filter_applies(search_filter, document):
                 ) and search) or doc_val == search
             elif isinstance(search, RE_TYPE) and isinstance(doc_val, (string_types, list)):
                 is_match = _regex(doc_val, search)
+            elif key in LOGICAL_OPERATOR_MAP:
```

Comment:
It's not sound to check for the `key` in a loop for which it's constant, so I've pushed a fix that only slightly modified the previous behavior (do not return True if it matches, just check the next key).